editing/edit-features-with-feature-linked-annotation/src/main/java/com.esri.samples.edit_features_with_feature_linked_annotation/EditFeaturesWithFeatureLinkedAnnotationSample.java

```diff
@@ -0,0 +1,350 @@
+    }
+  }
+
+  /**
+   * Selects the nearest feature, or move the point or polyline vertex to the given screen point.
+   *
+   * @param screenPoint the screen point at which to select a feature, or to move a feature to
+   */
+  private void selectOrMove(Point2D screenPoint) {
+
+    // if a feature hasn't been selected, select the feature
+    if (selectedFeature == null) {
+      identifyFeature(screenPoint);
+    } else {
+      // convert the screen point to a map point
+      Point mapPoint = mapView.screenToLocation(screenPoint);
+      // if the feature is a polyline, move the polyline
+      if (selectedFeatureIsPolyline) {
+        movePolylineVertex(mapPoint);
+      } else {
+        // if the feature is a point, move the point
+        movePoint(mapPoint);
+      }
+    }
+  }
+
+  /**
+   * Identifies a feature near the given screen point.
+   *
+   * @param screenPoint the screen point at which to identify a feature
+   */
+  private void identifyFeature(Point2D screenPoint) {
+
+    // clear any previously selected features
+    clearSelection();
+
+    // identify across all layers
+    ListenableFuture<List<IdentifyLayerResult>> identifyLayerResultsFuture = mapView.identifyLayersAsync(screenPoint, 10, false);
+    identifyLayerResultsFuture.addDoneListener(() -> {
+      try {
+        // get the list of results from the future
+        List<IdentifyLayerResult> identifyLayerResults = identifyLayerResultsFuture.get();
+        // for each layer from which an element was identified
+        for (IdentifyLayerResult layerResult : identifyLayerResults) {
+          // check if the layer is a feature layer, thereby excluding annotation layers
+          if (layerResult.getLayerContent() instanceof FeatureLayer) {
+            // get a reference to the identified feature
+            selectedFeature = (Feature) layerResult.getElements().get(0);
+            // check the geometry and select the feature
+            selectFeature(layerResult);
+            return;
+          }
+        }
+      } catch (Exception e) {
+        new Alert(Alert.AlertType.ERROR, "Error identifying the clicked feature.").show();
+      }
+    });
+  }
+
+  /**
+   * Checks if the identified feature is a straight polyline or a point, and select the feature.
+   * For a point feature, show a dialog to edit attributes. Future clicks will call move functions.
+   *
+   * @param layerResult the identify layer result from which to select a feature
+   */
+  private void selectFeature(IdentifyLayerResult layerResult) {
+
+    // if the selected feature is a polyline
+    if (selectedFeature.getGeometry().getGeometryType() == GeometryType.POLYLINE) {
+      // create a polyline builder from the selected feature
+      PolylineBuilder polylineBuilder = new PolylineBuilder((Polyline) selectedFeature.getGeometry());
+      // get a list of parts of the selected polyline
+      List<Part> parts = polylineBuilder.getParts();
+      parts.forEach(part -> {
```

Comment:
since we don't use the variable `parts` more than once, we can directly add the forEach like so:

```suggestion
      polylineBuilder.getParts().forEach(part -> {
```